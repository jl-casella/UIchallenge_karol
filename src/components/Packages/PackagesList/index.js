import PropTypes from 'prop-types'
import React from 'react'
import styled from 'styled-components'

import PackageTab from '../PackageTab'
import { Package } from 'types/Packages'

const PackagesListContainer = styled.ul`
  padding: 0;
  margin: 2px 0 10px;
  min-height: 60px;
  max-height: 30%;
  overflow-y: auto;
`

const PackagesList = ({
  packages,
  activePackageId,
  setActivePackageId,
  removePackage
}) => (
  <PackagesListContainer>
    {packages.map((p, i) => {
      return (
        <PackageTab
          key={p.id}
          number={i + 1}
          active={p.id === activePackageId}
          setActivePackage={() => setActivePackageId(p.id)}
          removePackage={() => removePackage(p.id)}
          removeEnabled={p.items.length === 0}
        />
      )
    })}
  </PackagesListContainer>
)

PackagesList.propTypes = {
  packages: PropTypes.arrayOf(Package).isRequired,
  activePackageId: PropTypes.string,
  setActivePackageId: PropTypes.func.isRequired,
  removePackage: PropTypes.func.isRequired
}

export default PackagesList
