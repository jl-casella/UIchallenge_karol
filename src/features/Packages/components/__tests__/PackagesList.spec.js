import { noop } from 'lodash'
import React from 'react'
import { render } from '@testing-library/react'

import PackagesList from '../PackagesList'

describe('PackagesList component test', () => {
  it('Should render correctly', () => {
    const packagesMock = [
      { id: 'test', items: [] },
      { id: 'test2', items: [] }
    ]

    const { container } = render(
      <PackagesList
        packages={packagesMock}
        activePackage={undefined}
        setActivePackageId={noop}
        removePackage={noop}
      />
    )

    expect(container.firstChild).toMatchSnapshot()
  })
})
