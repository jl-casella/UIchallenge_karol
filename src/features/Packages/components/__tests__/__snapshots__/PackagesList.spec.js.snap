// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PackagesList component test Should render correctly 1`] = `
<ul
  class="PackagesList__PackagesListContainer-sc-1qpl6ke-0 ELSMu"
>
  <li
    class="PackageTab__PackageTabContainer-sc-1wzierc-0 jolmYP"
  >
    <a>
      Package 
      1
    </a>
    <div
      class="PackageTab__RemoveButton-sc-1wzierc-1 jNsckS"
      role="button"
    >
      X
    </div>
  </li>
  <li
    class="PackageTab__PackageTabContainer-sc-1wzierc-0 jolmYP"
  >
    <a>
      Package 
      2
    </a>
    <div
      class="PackageTab__RemoveButton-sc-1wzierc-1 jNsckS"
      role="button"
    >
      X
    </div>
  </li>
</ul>
`;
